app/controllers/lightweight_activities_controller.rb

```diff
@@ -271,10 +271,14 @@ def resubmit_answers
 
   private
   def set_activity
-    if params[:activity_id]
-      @activity = LightweightActivity.find(params[:activity_id])
+    id = params[:id]
+    id = params[:activity_id] if params[:activity_id]
```

Comment:
It took me a second to realize the previous line was the fallback.  Maybe change this to ```id = params[:activity_id].present? ? params[:activity_id] : params[:id]```